pkg/cloud/services/compute/instance.go

```diff
@@ -420,6 +405,31 @@ func (s *Service) getImageIDByReference(ctx context.Context, k8sClient client.Cl
 	return nil, orcImage, nil
 }
 
+// Helper to resolve a flavor ID.
+// TODO: needs a breaking CRD change so it works like images.
+func (s *Service) GetFlavorID(flavorID, flavorName *string) (string, error) {
+	if flavorID != nil {
+		return *flavorID, nil
+	}
+
+	if flavorName == nil {
+		return "", fmt.Errorf("no flavors were found: no name set")
+	}
+
+	allFlavors, err := s.getComputeClient().ListFlavors()
+	if err != nil {
+		return "", err
+	}
+
+	for _, flavor := range allFlavors {
+		if flavor.Name == *flavorName {
+			return flavor.ID, nil
+		}
+	}
```

Comment:
I was going to write that client-side filtering is madness, but:
* That's what we do currently, presumably because
* Nova doesn't appear to support filtering by name anyway 🤯

Reply:
Oh it drives me mad, it really needs metadata of some variety so you can describe CPU microarchitectures, GPUs and other jazz.  All of which is really useful to an end user!